org.mixedrealitytoolkit.spatialmanipulation/BoundsControl/Visuals/SpatialManipulationReticle.cs

```diff
@@ -47,7 +47,23 @@ public void UpdateVisual(ReticleVisualUpdateArgs args)
                 }
                 else if (rayInteractor.interactablesHovered.Count > 0)
                 {
-                    RotateReticle(args.ReticleNormal, rayInteractor.interactablesHovered[0].transform);
+                    int interactableIndex = 0;
+
+                    List<IXRHoverInteractable> hoveredInteractables = rayInteractor.interactablesHovered;
```

Comment:
Should cache this to avoid the per-frame allocation

Reply:
There's already no per-frame allocation, luckily! The `.interactablesHovered` [property's documentation](https://docs.unity3d.com/Packages/com.unity.xr.interaction.toolkit@3.0/api/UnityEngine.XR.Interaction.Toolkit.Interactors.IXRHoverInteractor.html#UnityEngine_XR_Interaction_Toolkit_Interactors_IXRHoverInteractor_interactablesHovered) states:

> It is exposed as a List<T> rather than an IReadOnlyList<T> to avoid GC Allocations when enumerating the list.

and it calls into [a Unity CoreUtils helper](https://docs.unity3d.com/Packages/com.unity.xr.core-utils@2.4/api/Unity.XR.CoreUtils.Collections.HashSetList-1.html#Unity_XR_CoreUtils_Collections_HashSetList_1_AsList) that itself says:

> Exposes internal list without any allocation.

I also went ahead and profiled to verify:
![image](https://github.com/user-attachments/assets/97b84ff9-dbb7-41df-80d6-9aeec14226b7)

Reply:
Awesome, I learned something.